Contract update changes

bencalder opened the pull request:
@vvisigoth this PR addresses the following issues: #18 #19 #20 #21 #22 #23 #24 #25 #26 #27 #28

Commit:
changes to urbitwallet-master

bencalder committed:
New partials. Update all functions related to the state and lock time. Update variable names. Consolidate functions to match smart contracts.

vvisigoth commented:
Thanks Ben! I’ll take a look and get back to you with comments.

On Wed, May 30, 2018 at 9:52 PM Ben Calder <notifications@github.com> wrote:

> @vvisigoth <https://github.com/vvisigoth> this PR addresses the following
> issues: #18 <https://github.com/urbit/etherwallet/issues/18> #19
> <https://github.com/urbit/etherwallet/issues/19> #20
> <https://github.com/urbit/etherwallet/issues/20> #21
> <https://github.com/urbit/etherwallet/issues/21> #22
> <https://github.com/urbit/etherwallet/issues/22> #23
> <https://github.com/urbit/etherwallet/issues/23> #24
> <https://github.com/urbit/etherwallet/issues/24> #25
> <https://github.com/urbit/etherwallet/issues/25> #26
> <https://github.com/urbit/etherwallet/issues/26> #27
> <https://github.com/urbit/etherwallet/issues/27> #28
> <https://github.com/urbit/etherwallet/issues/28>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/urbit/etherwallet/pull/29
> Commit Summary
>
>    - changes to urbitwallet-master
>    - New partials. Update all functions related to the state and lock
>    time. Update variable names. Consolidate functions to match smart contracts.
>
> File Changes
>
>    - *M* app/includes/urbit.tpl
>    <https://github.com/urbit/etherwallet/pull/29/files#diff-0> (56)
>    - *M* app/partial/allowtransfer.html
>    <https://github.com/urbit/etherwallet/pull/29/files#diff-1> (2)
>    - *M* app/partial/buy.html
>    <https://github.com/urbit/etherwallet/pull/29/files#diff-2> (2)
>    - *A* app/partial/configurekeys.html
>    <https://github.com/urbit/etherwallet/pull/29/files#diff-3> (123)
>    - *M* app/partial/creategalaxy.html
>    <https://github.com/urbit/etherwallet/pull/29/files#diff-4> (26)
>    - *M* app/partial/launch.html
>    <https://github.com/urbit/etherwallet/pull/29/files#diff-5> (2)
>    - *A* app/partial/setspawnproxy.html
>    <https://github.com/urbit/etherwallet/pull/29/files#diff-6> (115)
>    - *A* app/partial/spawn.html
>    <https://github.com/urbit/etherwallet/pull/29/files#diff-7> (116)
>    - *M* app/partial/state.html
>    <https://github.com/urbit/etherwallet/pull/29/files#diff-8> (53)
>    - *M* app/scripts/controllers/tabsCtrl.js
>    <https://github.com/urbit/etherwallet/pull/29/files#diff-9> (7)
>    - *M* app/scripts/controllers/urbitCtrl.js
>    <https://github.com/urbit/etherwallet/pull/29/files#diff-10> (260)
>    - *M* app/scripts/main.js
>    <https://github.com/urbit/etherwallet/pull/29/files#diff-11> (28)
>    - *M* app/scripts/services/templateService.js
>    <https://github.com/urbit/etherwallet/pull/29/files#diff-12> (16)
>    - *M* dist/js/urbitwallet-master.js
>    <https://github.com/urbit/etherwallet/pull/29/files#diff-13> (99)
>    - *M* package-lock.json
>    <https://github.com/urbit/etherwallet/pull/29/files#diff-14> (8)
>
> Patch Links:
>
>    - https://github.com/urbit/etherwallet/pull/29.patch
>    - https://github.com/urbit/etherwallet/pull/29.diff
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/urbit/etherwallet/pull/29>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABiQ44HtxG9MFvv5mKz5TmCYRB1IiDaAks5t33b8gaJpZM4UUavM>
> .
>
-- 
—
~poldec-tonteg
http://urbit.org

vvisigoth committed:
fixed partial typo

vvisigoth committed:
changed const address

vvisigoth committed:
changed isSpawnProxy signature

vvisigoth committed:
use isACtive

vvisigoth committed:
user isOwner

vvisigoth committed:
spawn instead of launch

vvisigoth commented:
Spoke with Ben today, but summarizing here: We're still seeing some gasEstimation errors in doSpawn (and elsewhere). This is likely a dev environment thing and Ben is sorting it out.

bencalder committed:
Clean up in UI. Add a check to prevent planets from attempting to
spawn. Fix call to getOwner.

bencalder committed:
Fixed copy

bencalder committed:
Updated Vote UI

bencalder committed:
Removed unused action columns

bencalder committed:
Removed old partials

bencalder committed:
Removed call to unused function

bencalder committed:
Removed switch logic for state.

bencalder committed:
Descriptive variable names for clarity

bencalder committed:
Added check to prevent galaxies from being shown the deposit action

bencalder committed:
Added check for booted status for displaying correct action list

bencalder committed:
Updated contract addresses. Removed unused functions. Added call to
retrieve hasBeenBooted status for owned ships. Updated get functions to
new contract function names. Added checks to Spawn and Deposit
functions to verify booted status.

bencalder commented:
Gas estimation errors turned out to be caused by the smart contracts validating hasBeenBooted status.

bencalder committed:
Add call to wipe transaction when back button is clicked in action
partials

bencalder committed:
Remove unused partials

bencalder committed:
Moved if statement in pool function to prevent infinite loop

bencalder committed:
Reorder GET calls to group them

bencalder committed:
Add call to getEscapeRequest

bencalder committed:
Add call to getTransferringFor

bencalder committed:
Add call to approve

bencalder committed:
Add call to doSafeTransferFrom

bencalder committed:
Add ship validation to doApprove call

bencalder committed:
Add call to get canEscapeTo. Create checkCanEscapeTo function. Move checkHasBeenBooted to class function. Add validation logic to doEscape.
